portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.sps.data.MarkerClass;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+import java.util.List;
+/** Handles fetching and saving markers data. */
+@WebServlet("/markers")
+public class MarkerServlet extends HttpServlet {
+  /** Accepts a POST request containing a new marker. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)throws IOException {
+    /*Sends data of the new marker to the server */
+    double lat = Double.parseDouble(request.getParameter("lat"));
+    double lng = Double.parseDouble(request.getParameter("lng"));
+    String content = Jsoup.clean(request.getParameter("content"), Whitelist.none());
+    String title = Jsoup.clean(request.getParameter("title"),Whitelist.none());
+    MarkerClass marker = new MarkerClass(lat, lng, title, content);
+    storeMarker(marker);
+    
+    Gson gson = new Gson();
+    response.setContentType("application/json");
+    response.getWriter().println(gson.toJson(marker));
+    response.sendRedirect("/maps.html");
+  }
+  /** Stores a marker in Datastore. */
+  private void storeMarker(MarkerClass marker) {
```

Comment:
https://google.github.io/styleguide/javaguide.html#s3.4.2-ordering-class-contents

Let's have an ordering of our functions? Like grouping public together?

First, think only about the API (public and protected members) and the state (ordinary instance fields) of your type. Work out how to arrange those without worrying about any other members just yet.

We recommend a high-level ordering like this (notice that visible static fields and static methods are grouped together before instance fields):

1. Visible static fields (usually final)
2. Visible static methods
3. "Regular" instance fields (comprising object state, not derived)
4. Constructors (whether visible or not)
5. Visible instance methods

Within a category, keep the members grouped by subject area, and also try to put the more central and important features up high, and details that fewer readers will care about further down.